### PR TITLE
Add number of rows to SerializedPage

### PR DESCRIPTION
**Pull request by HolyLow**
Add optional number of rows field to SerializedPage.

Populated for pages produced by PartitionedOutput operator. 

SerializedPages produced by ExchangeSource may not have number of rows set.

Will be used in #7962

### CODE REVIEW

File: velox/exec/ExchangeQueue.h [hunk 1]
```diff
@@ -26,7 +26,8 @@ class SerializedPage {
   // Construct from IOBuf chain.
   explicit SerializedPage(
       std::unique_ptr<folly::IOBuf> iobuf,
-      std::function<void(folly::IOBuf&)> onDestructionCb = nullptr);
+      std::function<void(folly::IOBuf&)> onDestructionCb = nullptr,
+      std::optional<int64_t> rows = std::nullopt);
```
Comment (mbasmanova) [inline velox/exec/ExchangeQueue.h:30]: naming: numRows for consistency

Please, add @param comment to document the new parameter

File: velox/exec/ExchangeQueue.h [hunk 2]
```diff
@@ -35,6 +36,10 @@ class SerializedPage {
     return iobufBytes_;
   }
 
+  std::optional<int64_t> rows() const {
```
Comment (mbasmanova) [inline velox/exec/ExchangeQueue.h:39]: nit: numRows()

File: velox/exec/ExchangeQueue.h [hunk 3]
```diff
@@ -35,6 +36,10 @@ class SerializedPage {
...
+    return iobufRows_;
```
Comment (mbasmanova) [inline velox/exec/ExchangeQueue.h:40]: naming: numRows_

**Review by mbasmanova — COMMENTED** [review]
Thank. Look good over % some nits.

File: velox/exec/ExchangeQueue.h (hunk 1 above)
Comment (HolyLow) [reply velox/exec/ExchangeQueue.h]: Naming done, but it seems that the @param annotation is not used in the velox project.

File: velox/exec/ExchangeQueue.h (hunk 2 above)
Comment (HolyLow) [reply velox/exec/ExchangeQueue.h:39]: done.

File: velox/exec/ExchangeQueue.h (hunk 3 above)
Comment (HolyLow) [reply velox/exec/ExchangeQueue.h]: done.

**Comment by HolyLow** [comment]
@mbasmanova Any further comments?

**Review by mbasmanova — APPROVED** [review]
Looks good. Thanks.

**Pull request closed without merging**